File: 2017/day_05/day_05.py
import sys
data = []
for line in sys.stdin:
    data.append(int(line))

data_2 = data[:]

steps_1 = 0
i = 0
while i < len(data):
    a = data[i]
    data[i] += 1
    i += a
    steps_1 += 1

steps_2 = 0
i = 0
while i < len(data_2):
    a = data_2[i]
    if a >= 3:
        data_2[i] -= 1
    else:
        data_2[i] += 1
    i += a
    steps_2 += 1

print('Part One: {}\nPart Two: {}'.format(steps_1, steps_2))

File: 2017/day_05/day_05.cpp
#include <iostream>
#include <vector>
#include <string>

int main() {
    std::vector<int> data;

    std::string line;
    while(std::getline(std::cin, line)) data.push_back(std::stoi(line));
    std::vector<int> data_2 = data;


    int steps_1 = 0;
    int i = 0;
    while (i < data.size()) {
        int a = data[i];
        ++data[i];
        i += a;
        ++steps_1;
    }

    int steps_2 = 0;
    i = 0;
    while(i < data_2.size()) {
        int a = data_2[i];
        if (a >= 3) --data_2[i];
        else ++data_2[i];
        i += a;
        ++steps_2;
    }

    std::cout << "Part One: " << steps_1 << std::endl << "Part Two: " << steps_2 << std::endl;

    return 0;
}
